### PR TITLE
refactor(console): application details

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Refactor(console): application details:
- Merge the advanced settings tab into the settings page.
- Remove the advanced settings router and related components.
- Add a `<FormCard />` component to help to construct the common layout.
image

- Remove the margin-bottom from the `<TabNav />` to fit the new layout.

### Before
image

### After
application-details

### CODE REVIEW

File: packages/console/src/components/TabNav/index.module.scss [hunk 1]
```diff
@@ -3,5 +3,5 @@
 .nav {
   border-bottom: 1px solid var(--color-divider);
   display: flex;
-  margin: _.unit(1) 0;
+  margin: _.unit(1) 0 0;
```
Comment (simeng-li) [inline packages/console/src/components/TabNav/index.module.scss:6]: margin top？

File: packages/console/src/scss/_underscore.scss [hunk 2]
```diff
@@ -22,7 +22,7 @@
 }
 
 @mixin form-text-field {
-  width: dim.$form-text-field-width;
+  width: 100%;
```
Comment (simeng-li) [inline packages/console/src/scss/_underscore.scss:25]: just remove the width styling? most likely the default value will be the full width of the container.

File: packages/console/src/pages/ApplicationDetails/index.tsx [hunk 3]
```diff
@@ -200,51 +199,29 @@ const ApplicationDetails = () => {
...
+          <FormProvider {...formMethods}>
+            <form className={styles.formContent} onSubmit={onSubmit}>
+              <div className={styles.fieldsContent}>
+                <Settings data={data} />
+                <AdvancedSettings applicationType={data.type} oidcConfig={oidcConfig} />
+              </div>
+              <SubmitFormChangesActionBar
+                isOpen={isDirty}
+                isSubmitting={isSubmitting}
+                onDiscard={() => {
```
Comment (simeng-li) [inline packages/console/src/pages/ApplicationDetails/index.tsx:216]: onDiscard={reset}?

File: packages/console/src/scss/_underscore.scss (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/scss/_underscore.scss:25]: No, this style is used on other pages, before we replace them all, I think it's good to leave it here right now.
BTW, fields that have `display: inline-block` will need this style.
E.g. `<CopyToClipboard />`
image

File: packages/console/src/pages/ApplicationDetails/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/index.tsx:216]: Refactored here:
https://github.com/logto-io/logto/pull/2464

File: packages/console/src/components/TabNav/index.module.scss (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/components/TabNav/index.module.scss]: updated.

**Review by simeng-li — APPROVED** [review]
LGTM

File: packages/console/src/scss/_underscore.scss (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/scss/_underscore.scss:25]: Add a TODO or FIXME tag?

**Review by wangsijie — APPROVED** [review]

**Pull request merged**